Feature/celery

vlapey opened the pull request.

vlapey committed:
1. created celery.py file
2. listed celery in requirements.txt

vlapey committed:
1. added celery, redis to docker-compose
2. added celery config to settings.py

vlapey committed:
1. added simple testing task

vlapey committed:
added relation from car to other models

vlapey committed:
1. refactored docker-compose
2. added django-celery-beat to requirements.txt
3. added django-celery-beat to settings

vlapey committed:
added expose to docker-compose

vlapey committed:
created authentication app

vlapey committed:
1. created authentication app
2. created UserManager and extended User class in auth.models
3. changed naming from date_created to standart created_at
4. freezed requirements.txt
5. carried out logic of router registry in separate file
6. wrote serializer for User model
7. updated settings with django_extensions and removed usage of simple jwt
8. created urls in auth app
9. update config.urls so
10. wrote views for User model

vlapey committed:
added migrations to auth app

vlapey committed:
implemented registration via jwt

vlapey committed:
migrations

vlapey committed:
fixed problem with decoder in backends and auth models

vlapey committed:
1. removed admin role from core.enums
2. added role field to user model

vlapey committed:
changed messages to russian in backends

vlapey committed:
1. added host email string to env
2. created utils class in config

vlapey committed:
Merge branch 'feature/celery' into feature/email_confirm

vlapey committed:
Merge pull request #14 from vlapey/feature/email_confirm

Feature/email confirm

vlapey committed:
freezed requirements.txt

vlapey committed:
deleted unused references in env file

vlapey committed:
configured _init_ file in config

vlapey committed:
1. wrote celery beat schedule
2. added depends_on in docker-compose
3. added price field to car model
4. added car, price , discount fields in history models
5. added redis and celery config in settings
6. created car_task in cars folder

vlapey committed:
1. made migrations
2. created CarRandomizer class with static methods that randomize values for celery
3. added to enums CarEngineType, CarBrand, CarModel
4. changed engine field to choices() in Cars models
5. fixed issue in settings.py after merging
6 implemented car task for celery

Chaddi13 commented on authentication/models.py:
```diff
@@ -0,0 +1,71 @@
+
+class UserManager(BaseUserManager):
+    def create_user(self, username, email, password=None):
+        if username is None:
+            raise TypeError('Users must have a username.')
+
+        if email is None:
+            raise TypeError('Users must have an email address.')
+
```
Клиент, при регистрации, будет создавать только 1 юзера, поэтому нет смысла писать Users

Chaddi13 reviewed (changes requested):
Настрой celery, celery beat, flower, kafka
Добавь таски

vlapey committed:
1. renamed cars to car
2. put logic of car_specification and car randomizers into CarRandomizer class
3. restructured docker-compose file
4. deleted extra __str__ method in dealer model
5. dropped car_task and moved it to vendor_task
6. created delaer_task
7. created vendor_task
8. renamed CarsFilter to CarFilter

vlapey committed:
rebuilt migrations

vlapey committed:
1. put redis host, redis port to env
2. added flower to docker-compose
3. changed settings

vlapey committed:
1. updated env.local
2. changed name of db container docker-compose
3. refactored requirements.txt
4. added allowed hosts

vlapey committed:
completed docker compose (FLOWER IS NOT WORKING)

vlapey committed:
added command to flower in docker-compose, flower started to run

vlapey committed:
docker-compose ready project with working flower

vlapey committed:
1. refactored randomize_car_specification() method and added save method
2. made model non-unique field
3. refactored vendor task

vlapey committed:
1. refactored history models (made discount as foreign key)
2. made dealer task

vlapey committed:
refactored task logic

vlapey committed:
refactored models in history

vlapey committed:
deleted unused method

vlapey committed:
test commit, trying to get rid of car_specification

vlapey committed:
got rid of unused app Orders and changed settings.py

vlapey committed:
1. changed filters for car
2. got rid of orders in router_register func

vlapey committed:
1. dropped migrations because of conflicts

vlapey committed:
not full working version of a dealer task

vlapey committed:
1. implemented SpecsRandomizer
2. refactored dealer task

vlapey committed:
1. completed dealer_task without history records

vlapey committed:
1. changed models in history, that discount is just positive int
2. refactored dealers task so it's working and records history of transactions

vlapey committed:
1. renamed car_randomizer to car_service
2. put common methods from dealer_task into car_service
3. implemented customer_task

vlapey committed:
1. refactored car service to correctly check queries
2. added extra check on car.dealer and vendor

vlapey committed:
1. added extra checks

vlapey committed:
removed redundant imports
